### PR TITLE
Controller runtime 0.2.0 alpha0 vendor

### PR DESCRIPTION
**Pull request by corinnekrych**
**Description of the change:**

Adjust `List(gCtx goctx.Context, list runtime.Object, opts ...dynclient.ListOptionFunc` to match new [runtime-controller](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.2.0-alpha.0) signature.

**Motivation for the change:**

Make operator-sdk compatible with [runtime-controller](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.2.0-alpha.0)

### CODE REVIEW

**Comment by LiliC** [comment]
Hi @corinnekrych thanks so much for this PR! 🎉  

We might not be merging it soon as that would be breaking changes and we need to decide in which release we would want to have this. The release of controller-runtime is also an alpha one, so we might need to wait for a more stable release of that.

There also needs to be a migration doc for the changes, and some other things so maybe would make sense to have someone else from our team work on this with you if you are interested in this? Thanks again!

cc @joelanford @shawn-hurley @estroz @hasbro17 @AlexNPavel

**Comment by estroz** [comment]
I'd rather wait for a `v0.2.0` stable release. There's no urgency to upgrade for the next release.

File: internal/pkg/scorecard/resource_handler.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func getPodFromDeployment(depName, namespace string) (pod *v1.Pod, err error) {
 	// instead of the new one.
 	err = wait.PollImmediate(time.Second*1, time.Second*60, func() (bool, error) {
 		pods := &v1.PodList{}
-		err = runtimeClient.List(context.TODO(), &client.ListOptions{LabelSelector: set.AsSelector()}, pods)
+		err = runtimeClient.List(context.TODO(), pods, client.MatchingLabels(set))
```
Comment (camilamacedo86) [inline internal/pkg/scorecard/resource_handler.go]: Hi @corinnekrych,
In the https://github.com/operator-framework/operator-sdk/pull/1388/commits/f49484592eecb160259e2b162357c0982676fb92#diff-b38c855ac3ed0aa95294c855ea839f99R176 it was solved as follows. 
```go
err = runtimeClient.List(context.TODO(), pods, client.MatchingLabels(dep.Spec.Selector.MatchLabels))
```

File: pkg/ansible/proxy/proxy.go [hunk 2]
```diff
@@ -114,7 +117,7 @@ func CacheResponseHandler(h http.Handler, informerCache cache.Cache, restMapper
 					log.Error(err, "Unable to decode list options from request")
 					break
 				}
-				lo := client.InNamespace(r.Namespace)
+				lo := client.ListOptions{Namespace: r.Namespace}
```
Comment (camilamacedo86) [inline pkg/ansible/proxy/proxy.go]: here we can do just as follows. 

```go
lo := &client.ListOptions{}
lo.InNamespace(r.Namespace)
```

File: pkg/ansible/proxy/proxy.go [hunk 3]
```diff
@@ -128,7 +131,25 @@ func CacheResponseHandler(h http.Handler, informerCache cache.Cache, restMapper
...
+				s, err := fields.ParseSelector(lo.FieldSelector.String())
+				reqs := s.Requirements()
+				if len(reqs) > 0 {
+					for idx, reg := range reqs {
+						if reg.Operator == selection.Equals || reg.Operator == selection.DoubleEquals {
+							p = append(p, client.MatchingField(reqs[idx].Field, reqs[idx].Value))
+						}
+					}
+				}
+				err = informerCache.List(context.Background(), &un, p...)
```
Comment (camilamacedo86) [inline pkg/ansible/proxy/proxy.go]: Shows that it is not required. We can do as follows. 

```go				
err = informerCache.List(context.Background(), &un, client.UseListOptions(lo))
```

File: pkg/log/zap/logger.go [hunk 4]
```diff
@@ -34,7 +33,6 @@ func LoggerTo(destWriter io.Writer) logr.Logger {
 	syncer := zapcore.AddSync(destWriter)
 	conf := getConfig()
 
-	conf.encoder = &logf.KubeAwareEncoder{Encoder: conf.encoder, Verbose: conf.level.Level() < 0}
```
Comment (camilamacedo86) [inline pkg/log/zap/logger.go]: Following it fixed made in the PR: https://github.com/operator-framework/operator-sdk/pull/1388/commits/f49484592eecb160259e2b162357c0982676fb92#diff-4407315726717f30063d558928ad9188R37

```go
zapf "sigs.k8s.io/controller-runtime/pkg/log/zap"
	
conf.encoder = &zapf.KubeAwareEncoder{Encoder: conf.encoder, Verbose: conf.level.Level() < 0}
`

**Comment by camilamacedo86** [comment]
Hi @corinnekrych,

I checked this PR and we are addressing all here too: https://github.com/operator-framework/operator-sdk/pull/1388. This one has just 1 test/scenario which is falling. Wdyt over we move together with  https://github.com/operator-framework/operator-sdk/pull/1388 which shows closer to be finished with success?

**Comment by corinnekrych** [comment]
@LiliC I'm fine with it. Go ahead and close. I'll try yours.

**Comment by camilamacedo86** [comment]
HI @corinnekrych, 

I think that just the owners can close it. Could you please close this one? Let me know if you face some issue with https://github.com/operator-framework/operator-sdk/pull/1388.

Cheers,

**Comment by LiliC** [comment]
Closing in favour of #1388, thanks!

**Pull request closed without merging**